dataflow_lite/include/dataflow_lite/cloudwatch/cloudwatch_service.h

```diff
@@ -215,6 +218,14 @@ virtual T convertInputToBatched(const D &input) = 0;
  * Main workhorse thread that dequeues from the source and calls Task run.
  */
 void work() {
+  if (
+    (minimum_publish_period_ > std::chrono::milliseconds(0)) &&
+    (getCurrentTimestamp() - last_publish_time_ < minimum_publish_period_)
+  ) {
+    // A rate limit has been set, and this request has come in before we are allowed to publish again
```

Comment:
Warn that the rate limit has been exceeded?

Reply:
My only concern is that could be very noisy in the cases where it actually occurs - since this function gets called in a tight loop, there are likely to be a large number of throttles before every successful one, and this will continue happening until the logs have been drained - so we'd get big spam-line of "Throttling".

A utility from ROS that I might use is the throttled logger, but our AWS utilities don't seem to have that capability. You think it would be worth building a throttled logger?

Actually - a good way might be to only warn on the first throttle after a non-throttled call. That's what I'll do (I'll keep train of thought above for context)